keps/sig-api-machinery/20190228-priority-and-fairness.md

```diff
@@ -775,129 +788,188 @@ Some flow schemata.
 spec:
-  requestPriority:
-    name: system-top
-  match:
-  - and: # writes by admins (does this cover loopback too?)
-    - superSet:
-      field: groups
-      set: [ "system:masters" ]
+  matchingPrecedence: 1500
+  priorityLevelConfiguration:
+    name: exempt
+  rules:
+  - rule: # All resource requests from system-privileged group
+      verbs: ['*']
+      apiGroups: ['*']
+      resources: ['*']
+    subjects:
+    - kind: Group
+      name: "system:masters"
+  - rule: # Delegated auth[n/z] requests
+      verbs: ['*']
+      apiGroups: ['authentication.k8s.io', 'authorization.k8s.io']
+      resources: ['*']
+    subjects:
+    - kind: Group
+      name: "system:authenticated"     
+    - kind: Group
+      name: "system:unauthenticated"     
 ```
 
 ```yaml
 kind: FlowSchema
 meta:
-  name: system-high
+  name: nodes
 spec:
-  requestPriority:
+  matchingPrecedence: 2500
+  priorityLevelConfiguration:
     name: system-high
-  flowDistinguisher:
-    source: user
-    # no transformation in this case
-  match:
-  - and: # heartbeats by nodes
-    - superSet:
-      field: groups
-      set: [ "system:nodes" ]
-    - equals:
-      field: resource
-      value: nodes
-  - and: # kubelet and kube-proxy ops on system objects
-    - superSet:
-      field: groups
-      set: [ "system:nodes" ]
-    - equals:
-      field: namespace
-      value: kube-system
-  - and: # leader elections for system controllers
-    - patternMatch:
-      field: user
-      pattern: system:controller:.*
-    - inSet:
-      field: resource
-      set: [ "endpoints", "configmaps", "leases" ]
-    - equals:
-      field: namespace
-      value: kube-system
+  distinguisherMethod:
+    type: ByUser
+  rules:  
+  - rule: # All resource requests from node group
+      verbs: ['*']
+      apiGroups: ['*']
+      resources: ['*']
+    subjects:
+    - kind: Group
+      name: "system:nodes"
+  - rule: # All non-resource requests from node group
+      verbs: ['*']
+      nonResourceURLs: ['*']
+    subjects:
+    - kind: Group
+      name: "system:nodes"    
 ```
 
 ```yaml
 kind: FlowSchema
 meta:
-  name: system-low
+  name: leader-election
 spec:
-  matchingPriority: 900
-  requestPriority:
-    name: system-low
-  flowDistinguisher:
-    source: user
-    # no transformation in this case
-  match:
-  - and: # the garbage collector
-    - equals:
-      field: user
-      value: system:controller:garbage-collector
+  matchingPrecedence: 2500
+  priorityLevelConfiguration:
+    name: system-high
+  distinguisherMethod:
+    type: ByUser
+  rules:  
+  - rule: # Leader-election requests from kube-controller-manager/kube-scheduler
+      verbs: ['*']
+      apiGroups: ['coordination.k8s.io']
+      resources: ['leases']
+    subjects:
+    - kind: User
+      name: "system:kube-controller-manager"    
+    - kind: User
+      name: "system:kube-scheduler" 
 ```
 
 ```yaml
 kind: FlowSchema
 meta:
-  name: workload-high
+  name: kube-manager-controller
```

Comment:
This name covers only part of what is matched.  Did you mean to leave the scheduler in here?  It is also has a flow schema dedicated to it.

Reply:
ouch removed